lua/hop/hint.lua

```diff
@@ -5,14 +5,18 @@ local M = {}
 -- Regex hint mode.
 --
 -- Used to hint result of a search.
-function M.by_searching(pat, plain_search)
+function M.by_searching(pat, plain_search, same_line)
   if plain_search then
     pat = vim.fn.escape(pat, '\\/.$^~[]')
   end
   return {
     oneshot = false,
-    match = function(s)
-      return vim.regex(pat):match_str(s)
+    match = function(s, line_nr, cursor_pos)
+      if not same_line or line_nr + 1 == cursor_pos[1] then
+        return vim.regex(pat):match_str(s)
+      else
+        return 0, 0
+      end
```

Comment:
I struggle a bit to understand what this `same_line`  and the rest is for. Do you mind adding some comments?